src/faldbt/magics.py

```diff
@@ -0,0 +1,36 @@
+from IPython.core.magic import register_line_magic, needs_local_scope
+from fal import FalDbt
+
+
+@register_line_magic
+@needs_local_scope
+def init_fal(line="", local_ns={}):
+    """
+    Init fal magic variables. Must provide project_dir and profiles_dir.
+
+    Example: %init_fal project_dir=/my_project_dir profiles_dir=/my_profiles_dir
+    """
+    args = dict([arg.split("=") for arg in line.split()])
+    if not args.get("project_dir") or not args.get("profiles_dir"):
+        raise Exception(
+            """
+            Both project_dir and profiles_dir need to be provided:
+            Example: %init_fal project_dir=/my_project_dir profiles_dir=/my_profiles_dir
+            """
+        )
+
+    faldbt = FalDbt(args["project_dir"], args["profiles_dir"])
+
+    fal_globals = {
+        "ref": faldbt.ref,
+        "source": faldbt.source,
+        "write_to_source": faldbt.write_to_source,
```

Comment:
I think you are missing `faldbt.write_to_model`. But I wonder if passing `faldbt` as the global is enough?

```py
    local_ns.update(faldbt)
```

Although maybe `local_ns` *needs* a `dict` object.

Reply:
Good catch! Also missing `context`.

The idea is that the notebook runtime should be as similar as possible to fal runtime and in notebook runtime we don't have `context` and `write_to_model`. Should we have them? How would we accomplish it? What if one notebook is associated with two models?